crates/sargon/src/profile/v100/entity/has_entity_security_state.rs

```diff
@@ -0,0 +1,31 @@
+use crate::prelude::*;
+
+pub trait HasEntitySecurityState {
+    fn security_state(&self) -> EntitySecurityState;
+
+    fn virtual_hierarchical_deterministic_factor_instances(
+        &self,
+        filter_key_kind: impl Into<Option<CAP26KeyKind>>,
+    ) -> IndexSet<HierarchicalDeterministicFactorInstance> {
+        let mut factor_instances: IndexSet<
+            HierarchicalDeterministicFactorInstance,
+        > = IndexSet::new();
+        match self.security_state() {
+            EntitySecurityState::Unsecured { value } => {
+                factor_instances.insert(value.transaction_signing);
+                if let Some(authentication_signing) =
+                    value.authentication_signing
+                {
+                    factor_instances.insert(authentication_signing);
+                }
+            }
+        }
+        let Some(key_kind) = filter_key_kind.into() else {
+            return factor_instances;
+        };
+        factor_instances
+            .into_iter()
+            .filter(|fi| fi.key_kind() == Some(key_kind))
```

Comment:
oh! riiiiight.... this is a dangerous bug here! 

I forgot `key_kind()` method on `HDFactorInstance` returns `Option`, which it must do because of legacy `DerivatoinPath::BIP44Like`.

But we MUST include BIP44 factors as if they were `CAP26KeyKind::TransactionSigning` if we pass `filter_key_kind: CAP26KeyKind` to this `virtual_hierarchical_deterministic_factor_instances` method.

So we must massage this function / these types a little bit.

Start by adding a unit tests creating `HierarchicalDeterministicFactorInstance` with a `BIP44Like` DerivationPath`. Looks like we do not have a `legacy` account sample value, so start by adding that. And it is so important that we test Legacy accounts in general that I lean towards perhaps even changing our `Account::sample_other` to be a Legacy account - doing that will cascade A LOT. All our Profile sample values will have to be updated. So maybe we should defer it until Ghenadies https://github.com/radixdlt/sargon/pull/237 has been merged.

Anyway, you will need an Account with BIP44 `secp256k1` public key as VECI (Virtual Entity (Account) Creating (Factor) Instance), lets call it `legacy_account` here, and then call `virtual_hierarchical_deterministic_factor_instances` and specify that you care about `TransactionSigning`, the unit test should assert that `legacy_account` is indeed included. 

I think it might be as easy as updating the `key_kind` method https://github.com/radixdlt/sargon/blob/b9d99958230ffc6e10c64cd376679d2357ab7ec0/crates/sargon/src/profile/v100/factors/hierarchical_deterministic_factor_instance.rs#L79
and return `CAP26Path::Account` instead of `None` for `DerivationPath::BIP44Like`
Ensure to document this, something along those lines: "// It is not possible to create any BIP44 for Personas, all BIP44 key are used for Legacy Accounts for TransactionSigning. We note that we return a type called CAP26 for a non-CAP26-type such as BIP44 path, which is an inaccuracy. Alas, this feels OK and we can change `CAP26KeyKind` -> `KeyKind` later if we want."

you will need to update the name and expected value of this unit test:
https://github.com/radixdlt/sargon/blob/b9d99958230ffc6e10c64cd376679d2357ab7ec0/crates/sargon/src/profile/v100/factors/hierarchical_deterministic_factor_instance.rs#L222

Hopefully that is all you need to do.